embassy-stm32/src/timer/basic_timer.rs

```diff
@@ -0,0 +1,96 @@
+use crate::time::Hertz;
+use embassy::{util::Unborrow, waitqueue::WakerRegistration};
+
+use super::*;
+
+pub struct StateInner<'d, T: super::Basic16bitInstance> {
+    timer: T,
+    phantom: PhantomData<&'d T>,
+    update_waker: WakerRegistration,
+    ready: bool,
+}
+
+unsafe impl<'d, T: super::Basic16bitInstance> Send for StateInner<'d, T> {}
+unsafe impl<'d, T: super::Basic16bitInstance> Sync for StateInner<'d, T> {}
+
+impl<'d, T: super::Basic16bitInstance> PeripheralState for StateInner<'d, T>
```

Comment:
It'd be nice to not use PeripheralMutex, storing the `WakerRegistration` in a `static` instead.